modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaUpdateStatusMethodCheck.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaUpdateStatusMethodCheck
+	extends BaseUpgradeMatcherReplacementCheck {
+
+	@Override
+	protected String formatMatcherIteration(
+		String content, String newContent, Matcher matcher) {
+
+		String methodCall = JavaSourceUtil.getMethodCall(
+			content, matcher.start());
+
+		String variable = matcher.group(2);
+
+		if (!methodCall.contains("UserLocalServiceUtil") &&
+			!hasClassOrVariableName(
+				"UserLocalService", content, content, variable) &&
+			!hasClassOrVariableName(
+				"UserService", content, content, variable)) {
+
+			return newContent;
+		}
+
+		List<String> parameterList = JavaSourceUtil.getParameterList(
+			methodCall);
+
+		if (parameterList.size() != 2) {
+			return content;
+		}
+
+		if (!hasParameterTypes(
+				content, content, ArrayUtil.toStringArray(parameterList),
+				new String[] {"long", "int"})) {
+
+			addMessage(
+				content,
+				StringBundler.concat(
+					"Could not resolve types of updateStatus method. The ",
+					"method signature has changed to updateStatus(long userId,",
+					"int status, ServiceContext serviceContext). Fill the new ",
+					"parameter manually."));
+		}
+
+		StringBundler sb = new StringBundler(4);
+
+		sb.append(JavaSourceUtil.getIndent(methodCall));
+		sb.append("ServiceContext serviceContextThreadLocal = ");
+		sb.append("ServiceContextThreadLocal.getServiceContext();\n\n");
+		sb.append(matcher.group(1));
+
+		String newMethod = JavaSourceUtil.addMethodNewParameters(
+			parameterList.size(), methodCall, variable,
+			Arrays.asList("(ServiceContext)serviceContextThreadLocal.clone()"),
```

Comment:
@TamyrisTorres, if you do `new String[] {"(ServiceContext)serviceContextThreadLocal.clone()"}` you don't need to call another class.

Reply:
I use the Arrays class because the addMethodNewParameters method expects a List.